File: easy/1603_design_parking_system/solution.py
class ParkingSystem:
    def __init__(self, big: int, medium: int, small: int):
        self.__parking = [big, medium, small]

    def addCar(self, car_type: int) -> bool:
        if not self.__parking[car_type - 1]:
            return False

        self.__parking[car_type - 1] -= 1
        return True

File: easy/1603_design_parking_system/solution.cpp
#include <vector>

class ParkingSystem {
private:
  std::vector<int> parking;

public:
  ParkingSystem(int big, int medium, int small) {
    parking = {big, medium, small};
  }

  bool addCar(int carType) {
    if (parking[carType - 1] > 0) {
      --parking[carType - 1];
      return true;
    }

    return false;
  }
};
